Feature/new egress improvements (#13)
on-ramp/api-doc: source/includes/_egress.md

### Request JSON Fields

> Field             |   Type          | Description                                                                                                                          | Required 

Reviewer: > Required

:+1: good idea adding this!

----------------- | --------------- | ------------------------------------------------------------------------------------------------------------------------------------ | --------

> fiat_amount       | Integer         | Eur amount to be paid denominated in cents.                                                                                          | Yes

Reviewer: It is no longer Eur but whatever `fiat_currency` is specified. This could be misleading, as if one specified Usd he would not be sure whether to directly send the usd amount or the equivalent to the usd amount in eur.

Author: Made a change there. Check it please!

> fiat_currency     | String          | `"EUR"` or `"USD"`, depending what is available on the merchants account settings.                                                   | Yes 

Reviewer: > Unit currency for `fiat_amount` field in capital letters (i.e `"EUR"` or `"USD"` ), what currencies are allowed on this field depends on merchant settings.

?

Author: See if now it makes more sense.

payment_ack_url   | String          | Merchant callback endpoint to confirm egress transaction. It should be a complete, well formed, url.                                 | Yes

> user_redirect_url | String          | Where to redirect the user after the egress has been confirmed. It should be a complete, well formed, url.                           | Yes

Reviewer: We are probably start to use many more url on several places, like for the new egress flow, but also for example for the redirect tile.

What about changing the type `String` to `Url` (or `Callback` or similar) and later somewhere specify what we understand for a field of type "url"/"callback" maybe with some example of what would be correct and incorrect url to us?

Author: Changed to Url field. I don't think we need though to specify how a callback should be as I understand Url is self explanatory.

timeout_in_sec    | Integer         | When to expire the link if unused.                                                                                                   | Yes
offer_skin        | Egress Skin     | Specify how the offer should be displayed to the user.                                                                               | Yes
billing_details   | Billing Details | User billing details. Please, notice how **not** all parameters inside this json object are required except `merchant_customer_id`.  | Yes

### Egress Skin

Field             |   Type      | Description                                               | Required
----------------- | ----------- | --------------------------------------------------------- | --------
title             | string      | Short string containing merchant's or redemption's name.  | Yes

> image             | url         | image to stylized the offer.                              | Yes

Reviewer: Here we are already using `url` instead of `string`, but somewhere else we don't (I know this is not from the PR but from a previous one). Also another inconsistency, probably it would look better if we stick to a consistent capitalization for both the `Field` field and the `Type` field.

Author: Made those changes too.

member_since          | Date        | The date since that user is a member of yours.                        | No
merchant_customer_id  | String      | The merchant customer id.                                             | Yes

### Callback notifying about user registered or didn't do it before the expiring date

> Either once the user signs up with **ON/RAMP** or after some defined expiration time passes, **ON/RAMP** will notify the merchant about if they provided a valid `user_ready_url` parameter.

Reviewer 2: ```suggestion
Either once the user signs up with **ON/RAMP** or after an expiration time passes, **ON/RAMP** will notify the merchant about if they provided a valid `user_ready_url` parameter.
```

Author: Done!
